src/Spice86.Core/Emulator/InterruptHandlers/Bios/BiosMemoryInt12Handler.cs

```diff
@@ -0,0 +1,44 @@
+namespace Spice86.Core.Emulator.InterruptHandlers.Bios;
+
+using Spice86.Core.Emulator.Callback;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Core.Emulator.VM;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+/// <summary>
+/// BIOS service for knowing the size of conventional memory on startup.
+/// See: https://stanislavs.org/helppc/int_12.html
+/// </summary>
+public class BiosMemoryInt12Handler : InterruptHandler {
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="machine">The emulator machine.</param>
+    /// <param name="loggerService">The logger service implementation.</param>
+    public BiosMemoryInt12Handler(Machine machine, ILoggerService loggerService) : base(machine, loggerService) {
+        _memory.SetUint16(MemoryUtils.ToPhysicalAddress(MemoryMap.BiosMemorySizeSegment, 0), 640);
```

Comment:
If a 1k EBDA (Extended Bios Data Area, used by Mouse driver) is in use, this will report `639`